exporter/http_test.go

```diff
@@ -288,6 +288,56 @@ func TestHttpHandlers(t *testing.T) {
 	}
 }
 
+func TestReloadHandlers(t *testing.T) {
+	// if os.Getenv("TEST_PWD_REDIS_URI") == "" {
+	// 	t.Skipf("TEST_PWD_REDIS_URI not set - skipping")
+	// }
+
+	eWithPwdfile, _ := NewRedisExporter(os.Getenv("TEST_PWD_REDIS_URI"), Options{Namespace: "test", Registry: prometheus.NewRegistry(), RedisPwdFile: "../contrib/sample-pwd-file.json"})
+	ts := httptest.NewServer(eWithPwdfile)
+	defer ts.Close()
+
+	for _, tst := range []struct {
+		e    *Exporter
+		path string
+		want string
+	}{
+		{
+			path: "/-/reload",
+			want: `ok`,
+		},
+	} {
+		t.Run(fmt.Sprintf("path: %s", tst.path), func(t *testing.T) {
+			body := downloadURL(t, ts.URL+tst.path)
+			if !strings.Contains(body, tst.want) {
+				t.Fatalf(`error, expected string "%s" in body, got body: \n\n%s`, tst.want, body)
+			}
+		})
+	}
+
+	eWithnoPwdfile, _ := NewRedisExporter(os.Getenv("TEST_PWD_REDIS_URI"), Options{Namespace: "test", Registry: prometheus.NewRegistry()})
```

Comment:
You can cover the third case by using [sample-pwd-file.json-malformed](https://github.com/oliver006/redis_exporter/blob/master/contrib/sample-pwd-file.json-malformed) as the password file and then check that you get the expected error message when it fails to load passwords from that file.